src/silx/gui/plot/actions/control.py

```diff
@@ -330,14 +331,26 @@ def __init__(self, plot, parent=None):
         self.plot.sigActiveImageChanged.connect(self._updateColormap)
         self.plot.sigActiveScatterChanged.connect(self._updateColormap)
 
-    def setColorDialog(self, colorDialog):
+    def setColormapDialog(self, colormapDialog):
         """Set a specific color dialog instead of using the default dialog."""
-        assert(colorDialog is not None)
-        assert(self._dialog is None)
-        self._dialog = colorDialog
-        self._dialog.visibleChanged.connect(self._dialogVisibleChanged)
+        assert(colormapDialog is not None)
+        if self._dialog is not None:
+            self._dialog.visibleChanged.disconnect(self._dialogVisibleChanged)
+
+        self._dialog = colormapDialog
+        self._dialog.visibleChanged.connect(self._dialogVisibleChanged, qt.Qt.UniqueConnection)
         self.setChecked(self._dialog.isVisible())
 
+    @deprecated(replacement="setColormapDialog", since_version="2.0")
+    def setColorDialog(self, colorDialog):
+        self.setColormapDialog(colormapDialog=colorDialog)
+
+    def getColormapDialog(self):
+        if self._dialog is None:
+            self._dialog = self._createDialog(self.plot)
+            self._dialog.visibleChanged.connect(self._dialogVisibleChanged)
```

Comment:
It sounds an unexpected place to connect the signal, it would be best in `_createDialog`, but I guess there is a reason for not doing it there.

Reply:
I agree. But it cannot be done on the `_createDialog` which is static. And we want to build the dialog during runtime. So for now sounds like the safer solution but I agree this is weird.

Reply:
Yes, let's keep it as it is